pubsub/streaming-analytics/README.md

```diff
@@ -142,25 +145,27 @@ The following example will run a streaming pipeline. It will read messages from
 + `--output_path`: sets the output GCS path prefix to write files to
 + `--runner`: specifies the runner to run the pipeline, if not set to `DataflowRunner`, `DirectRunner` is used
 + `--window_size [optional]`: specifies the window size in minutes, defaults to 1.0
++ `--num_shards`: sets the number of shards when writing windowed elements to GCS, defaults to 1.
 + `--temp_location`: needed for executing the pipeline
 
 ```bash
 python PubSubToGCS.py \
-  --project=$PROJECT_NAME \
+  --project=$PROJECT_ID \
   --region=us-central1 \
-  --input_topic=projects/$PROJECT_NAME/topics/$TOPIC_NAME \
-  --output_path=gs://$BUCKET_NAME/samples/output \
+  --input_topic=projects/$PROJECT_ID/topics/$TOPIC_ID \
+  --output_path=gs://$BUCKET_ID/samples/output \
   --runner=DataflowRunner \
-  --window_size=2 \
-  --temp_location=gs://$BUCKET_NAME/temp
+  --window_size=1 \
+  --num_shards=2 \
```

Comment:
I don't have a strong opinion about this, but it might be a good idea to _not_ include `--num_shards` here since it slightly complicates how we run the sample and it's a technical word not everyone knows.

We could have like an info box saying that you might see more than one file per window, and that's because the elements in the windows might be processed in parallel, and that you can control how many files you want by specifying the number of shards, and that each shard corresponds to one file. This might be a good place to explain what shards are (like the number of files per window), why you would use them, what it defaults to, and how to choose a number (it would be a function of the number of messages per second, the window size, and the approximate number of messages you want per file).

Reply:
Commented this one out in the README.